src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.util;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.model.Template;
+import org.opensearch.flowframework.model.Workflow;
+import org.opensearch.flowframework.model.WorkflowNode;
+
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.amazonaws.encryptionsdk.AwsCrypto;
+import com.amazonaws.encryptionsdk.CommitmentPolicy;
+import com.amazonaws.encryptionsdk.CryptoResult;
+import com.amazonaws.encryptionsdk.jce.JceMasterKey;
+
+import static org.opensearch.flowframework.common.CommonValue.CREDENTIAL_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.MASTER_KEY;
+import static org.opensearch.flowframework.common.CommonValue.MASTER_KEY_INDEX;
+
+/**
+ * Encryption utility class
+ */
+public class EncryptorUtils {
+
+    private static final Logger logger = LogManager.getLogger(EncryptorUtils.class);
+
+    private ClusterService clusterService;
+    private Client client;
+    private String masterKey;
+
+    /**
+     * Instantiates a new EncryptorUtils object
+     * @param clusterService the cluster service
+     * @param client the node client
+     */
+    public EncryptorUtils(ClusterService clusterService, Client client) {
+        this.masterKey = null;
+        this.clusterService = clusterService;
+        this.client = client;
+    }
+
+    /**
+     * Sets the master key
+     * @param masterKey the master key
+     */
+    protected void setMasterKey(String masterKey) {
```

Comment:
Not sure the purpose of `protected` here (and many times below), as that implies subclassing.

If it's to enable testing we should use the default (package private) visibility. 

https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html